File: src/ai/flows/predict_workout_skipping.py
"""
This module defines a function for predicting workout skipping and sending motivational nudges.

- `predict_workout_skipping`: A function that predicts workout skipping and triggers motivational nudges.
- `PredictWorkoutSkippingInput`: The input type for the `predict_workout_skipping` function.
- `PredictWorkoutSkippingOutput`: The return type for the `predict_workout_skipping` function.
"""

from pydantic import BaseModel, Field
from typing import Optional

class PredictWorkoutSkippingInput(BaseModel):
    behavioral_data: str = Field(..., description="Historical data about the user's workout habits.")
    current_motivation_level: str = Field(..., description="The user's current self-reported motivation level.")
    schedule: str = Field(..., description="The user's current workout schedule.")

class PredictWorkoutSkippingOutput(BaseModel):
    skip_prediction: bool = Field(..., description="Whether or not the user is predicted to skip their next scheduled workout.")
    motivation_nudge: str = Field(..., description="A motivational message to encourage the user to stick to their workout schedule.")
    adjusted_schedule_suggestion: Optional[str] = Field(None, description="A suggested adjustment to the workout schedule.")

def predict_workout_skipping(input: PredictWorkoutSkippingInput) -> PredictWorkoutSkippingOutput:
    """
    Predicts workout skipping and sends motivational nudges.

    Args:
        input (PredictWorkoutSkippingInput): The input data for generating predictions.

    Returns:
        PredictWorkoutSkippingOutput: The prediction and motivational nudge.
    """
    # Simulate AI-based prediction logic
    return PredictWorkoutSkippingOutput(
        skip_prediction=True,
        motivation_nudge="Don't skip your workout today! Remember your goals.",
        adjusted_schedule_suggestion="Consider a lighter workout if you're feeling unmotivated."
    )

# Example usage
if __name__ == "__main__":
    example_input = PredictWorkoutSkippingInput(
        behavioral_data="Works out 3 times a week, skips Mondays.",
        current_motivation_level="low",
        schedule="Monday, Wednesday, Friday"
    )
    result = predict_workout_skipping(example_input)
    print(result.json(indent=4))

File: src/ai/flows/predict-workout-skipping.ts
export type PredictWorkoutSkippingInput = {
  behavioral_data: string;
  current_motivation_level: string;
  schedule: string;
};

export type PredictWorkoutSkippingOutput = {
  skip_prediction: boolean;
  motivation_nudge: string;
  adjusted_schedule_suggestion?: string;
};

export async function predictWorkoutSkipping(input: PredictWorkoutSkippingInput): Promise<PredictWorkoutSkippingOutput> {
  try {
    const response = await fetch('http://localhost:10001/api/predict-workout-skipping', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(input),
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    console.error('Error calling predict workout skipping API:', error);
    // Fallback to sample data if API is not available
    return {
      skip_prediction: false,
      motivation_nudge: "Keep up the great work! You're doing amazing.",
      adjusted_schedule_suggestion: "Consider adding variety to your routine."
    };
  }
}
